src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShardSyncer.java

```diff
@@ -149,19 +161,40 @@ private static synchronized void syncShardLeases(IKinesisProxy kinesisProxy,
 
     /** Helper method to detect a race condition between fetching the shards via paginated DescribeStream calls
      * and a reshard operation.
+     * @param inconsistentShardIds
+     * @throws KinesisClientLibIOException
+     */
+    private static void assertAllParentShardsAreClosed(Set<String> inconsistentShardIds)
+        throws KinesisClientLibIOException {
+        if (!inconsistentShardIds.isEmpty()) {
+            String ids = "";
```

Comment:
You can use [StringUtils#join](https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html) here instead of manually building the string.

Reply:
Updated.  I don't normally use Java as has no doubt been apparent. :-)